optimum/habana/diffusers/pipelines/ddpm/pipeline_ddpm.py

```diff
@@ -0,0 +1,195 @@
+
+        Example:
+
+        ```py
+        >>> from diffusers import DDPMPipeline
+
+        >>> # load model and scheduler
+        >>> pipe = DDPMPipeline.from_pretrained("google/ddpm-cat-256")
+
+        >>> # run pipeline in inference (sample random noise and denoise)
+        >>> image = pipe().images[0]
+
+        >>> # save image
+        >>> image.save("ddpm_generated_image.png")
+        ```
+
+        Returns:
+            [`~pipelines.ImagePipelineOutput`] or `tuple`:
+                If `return_dict` is `True`, [`~pipelines.ImagePipelineOutput`] is returned, otherwise a `tuple` is
+                returned where the first element is a list with the generated images
+        """
+        # Sample gaussian noise to begin loop
+        if isinstance(self.unet.config.sample_size, int):
+            image_shape = (
+                batch_size,
+                self.unet.config.in_channels,
+                self.unet.config.sample_size,
+                self.unet.config.sample_size,
+            )
+        else:
+            image_shape = (batch_size, self.unet.config.in_channels, *self.unet.config.sample_size)
+
+        if self._device.type == "mps":
+            # randn does not work reproducibly on mps
+            image = randn_tensor(image_shape, generator=generator)
+            image = image.to(self.device)
+        elif self._device.type =="hpu": # Patch random tensor
+            image = torch.randn(image_shape, generator=generator, device="cpu")
+            image = image.to(self._device)
+        else:
+            image = randn_tensor(image_shape, generator=generator, device=self.device)
+
+        # set step values
+        self.scheduler.set_timesteps(num_inference_steps, device="cpu") # Patch timesteps
+        timesteps = self.scheduler.timesteps.to(self._device)
+        if isinstance(self.scheduler, DDIMScheduler):
+            self.scheduler.reset_timestep_dependent_params()
+        num_inference_steps = [1] * len(self.scheduler.timesteps)
+
+        #Gaudi Patch
+        if not hasattr(self, 'timestep_unet'):
+            self.timestep_unet = copy.deepcopy(self.unet)
+            logger.info('Prepared a Unet for timestep calculations')
+
+        if self.use_hpu_graphs:
+           self.unet = self.ht.hpu.wrap_in_hpu_graph(self.unet, disable_tensor_cache=True)
+           self.timestep_unet = self.ht.hpu.wrap_in_hpu_graph(self.timestep_unet, disable_tensor_cache=True)
+
+        if self.use_habana:
+            self.unet = self.unet.to(self._device)
+            self.timestep_unet = self.timestep_unet.to(self._device)
+
+        start_time = time.time()
+        for i in self.progress_bar(num_inference_steps):
+            timestep = timesteps[0]
+            timesteps = torch.roll(timesteps, shifts=-1, dims=0)
+            emb = self.timestep_unet(image, timestep, True, None)
+
+            # 1. predict noise model_output
```

Comment:
Minor: You could capitalize the first letter of every comment like on line 188 to be consistent